packages/commerce/frontend/src/lib/stores/cart-store.ts

```diff
@@ -0,0 +1,60 @@
+import { Product } from '@/hooks/useProductsQuery'
+import { resetStore, valtioPersist } from '@/lib/valtio-persist.ts'
+import { watch } from 'valtio/utils'
+
+interface CartItem extends Product {
+  quantity: number
+}
+
+interface CartState {
+  items: CartItem[]
+  totalItems: number
+}
+
+const state = {
+  items: [],
+  totalItems: 0
+}
+
+export const cartStore = valtioPersist<CartState>('cart-store', state)
+
+export function addToCart(product: Product, quantity: number = 1): void {
```

Comment:
Valtio documentation suggests to have actions stored like this, since it's the recommended way for code splitting. We can group them into a class as well.